### PR TITLE
New Source: Gridly

### PR DESCRIPTION
**Pull request by hntan**
## What
Add source Gridly. [Gridly](https://www.gridly.com/) is a spreadsheet-based content management. It helps to manage your games or applications content in one place.

## How
Using [Gridly API](https://www.gridly.com/docs/api/#record). This connector contains record streams: `Record` which is the main data on Gridly. The connector requires two parameters: `api_key` and `grid_id` (or `branch_id`). 

## Recommended reading order
1. `spec.yaml`
2. `source.py`
3. `helpers.py`

## 🚨 User Impact 🚨
There is no change to the existing code

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector: Gridly</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

Screen Shot 2022-10-25 at 2 34 20 PM

Screen Shot 2022-10-25 at 2 34 29 PM

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-gridly/integration_tests/catalog.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "streams": [
+    {
+      "name": "TODO fix this file",
+      "supported_sync_modes": ["full_refresh", "incremental"],
+      "source_defined_cursor": true,
+      "default_cursor_field": "column1",
+      "json_schema": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gridly/integration_tests/catalog.json:9]: Remove this file.

File: airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  required:
+    - api_key
+    - view_id
+  properties:
+    api_key:
+      type: string
+      description: API Key
+    view_id:
+      type: string
+      description: View ID
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml:15]: I'm looking the API for Gridly and there are ways to retrieve all views and use a substream to read individual views without informing in the config.
Also there are a lot of other endpoints you can implement for this connector:
* database
* grid
* ...

Let me know if you need any assistance to implement the review I requested.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-gridly/integration_tests/catalog.json (hunk 1 above)
Comment (hntan) [reply airbyte-integrations/connectors/source-gridly/integration_tests/catalog.json:9]: Thanks you. I view remove the file.

File: airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml (hunk 2 above)
Comment (hntan) [reply airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml]: I updated the config, users now can specify a grid id and they can select which views to sync records

**Comment by hntan** [comment]
Integration test passed:

Screen Shot 2022-10-25 at 2 34 29 PM
Screen Shot 2022-10-25 at 2 34 20 PM

File: airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml (hunk 2 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml]: Ok Thanks!

File: airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      type: string
+      title: API Key
+      airbyte_secret: true
+    grid_id:
+      type: string
+      title: Grid ID
+      description: ID of a grid, or can be ID of a branch
+
+
+
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml:19]: Correct EOF

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments, almost done!

File: airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+documentationUrl: https://docsurl.com
```
[the remark this answers is not in the document]
Comment (hntan) [reply airbyte-integrations/connectors/source-gridly/source_gridly/spec.yaml]: Thanks! updated

File: airbyte-integrations/connectors/source-gridly/source_gridly/helpers.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            elif e.response.status_code == 404:
+                raise Exception(f"Grid id '{grid_id}' not found")
+            else:
+                raise Exception(f"Error getting listing views of grid '{grid_id}'")
+
+        return response.json()
+
+    @staticmethod
+    def get_grid(auth: TokenAuthenticator, grid_id: str) -> Dict[str, Any]:
+        url = f"https://api.gridly.com/v1/grids/{grid_id}"
```
[the remark this answers is not in the document]
Comment (hntan) [reply airbyte-integrations/connectors/source-gridly/source_gridly/helpers.py]: Thanks! updated.

**Comment by marcosmarxm** [comment]
@hntan can you share the output of `./gradlew airbyte-integrations:connectors:source-gridly:integrationTest`?

**Comment by hntan** [comment]
> @hntan can you share the output of `./gradlew airbyte-integrations:connectors:source-gridly:integrationTest`?

Hi @marcosmarxm, below is the result of integrationTest

Screen Shot 2022-10-27 at 12 12 56 PM
Screen Shot 2022-10-27 at 12 13 24 PM
Screen Shot 2022-10-27 at 12 13 35 PM

**Review by marcosmarxm — APPROVED** [review]
thanks @hntan

**Pull request merged**